pkg/apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -513,59 +504,36 @@ func (apimanager *APIManager) setSystemSpecDefaults() (bool, error) {
 	return changed, nil
 }
 
-func (apimanager *APIManager) setSystemFileStorageSpecDefaults() (bool, error) {
+func (apimanager *APIManager) setSystemFileStorageSpecDefaults() bool {
 	changed := false
 	systemSpec := apimanager.Spec.System
 
-	defaultFileStorageSpec := &SystemFileStorageSpec{
-		PVC: &SystemPVCSpec{
-			StorageClassName: nil,
-		},
-	}
-
 	if systemSpec.FileStorageSpec == nil {
-		systemSpec.FileStorageSpec = defaultFileStorageSpec
+		systemSpec.FileStorageSpec = &SystemFileStorageSpec{}
 		changed = true
-		return changed, nil
 	}
 
-	if systemSpec.FileStorageSpec.PVC != nil && systemSpec.FileStorageSpec.S3 != nil {
-		return changed, fmt.Errorf("Only one FileStorage can be chosen at the same time")
-	}
-
-	if systemSpec.FileStorageSpec.PVC == nil && systemSpec.FileStorageSpec.S3 == nil {
-		systemSpec.FileStorageSpec.PVC = defaultFileStorageSpec.PVC
+	// No default values for S3
+	// PVC
+	if systemSpec.FileStorageSpec.PVC == nil {
```

Comment:
If I understand correctly, this makes the  systemSpec.FileStorageSpec.PVC to be always defined.
This means that the resources related to PVC are always created now, which is not what we desire. We want to create them only when PVC is selected, not when S3 is selected.
What defines whether one of the storage options is used or not is whether it is nil or not.

Reply:
good catch!, fixed